### ISSUE TITLE
nhost ingress gRPC and mTLS support

### ISSUE THREAD

**Issue by xmlking** [nhost/nhost#2889]
As discussed on [discord](https://discord.com/channels/552499021260914688/1276029566913216524)

We have an use case where we want to run **gRPC** service on  `nhost Run`  and have client app (a daemon process running on user computer) communicate to server over [mTLS](https://grpc.io/docs/guides/auth/).

Will nhost cloud ingress allows passthrough `gRPC` /  `HTTP/2` connection to  nhost Run?
Will nhost cloud ingress allows passthrough mTLS client cert to backend  nhost run service?

if not, please help supporting this kind of workloads on nhost.

**Comment by dbarrosop** [comment]
> Will nhost cloud ingress allows passthrough gRPC /  HTTP/2 connection to nhost Run?

I thought we did but I did some verification and we need a small change. However, because of some internal reasons this will only work via a custom domain.

> Will nhost cloud ingress allows passthrough mTLS client cert to backend nhost run service?

Sort of. What we were thinking in doing is the following:

1. Terminate TLS, as usual, in our infrastructure
2. Perform the client certificate authentication in our own infrastructure
3. If the client succeeds, forward the headers:
- `ssl-client-issuer-dn`: For example, "CN=My CA"
- `ssl-client-subject-dn`: For example: "CN=My Client"
- `ssl-client-cert`: The full client certificate in PEM format. For example: `-----BEGIN%20CERTIFICATE-----%0A...---END%20CERTIFICATE-----%0A`

So basically we'd do the client certificate validation for you but forward to your backend all the information about the client certificate in case you need it (i.e. to identify the user). No need for configuring TLS on your backend.

Another thing to take into account is that if the client also needs to authenticate the server we have two options here:

1. Either you use the certificate we provide by default in our custom DNS solution
2. Allow you to upload your own cert

Let us know if this is suitable and which features are a must so we can prioritize each one accordingly (if you have a specific timeframe in mind that'd be ideal too).

**Comment by dbarrosop** [comment]
This is ready:

https://docs.nhost.io/platform/tls
https://docs.nhost.io/guides/run/networking#grpc

The CLI isn't fixed yet as that will require more thorough changes and we don't have the cycles for that yet, i just wanted to unblock the cloud. Please, test it and let us know how it goes.

**Comment by xmlking** [comment]
thanks a lot. thats all I needed
